crates/s3s/src/ops/generated.rs

```diff
@@ -552,14 +552,16 @@ impl super::Operation for CompleteMultipartUpload {
     async fn call(&self, s3: &Arc<dyn S3>, req: &mut http::Request) -> S3Result<http::Response> {
         let input = Self::deserialize_http(req)?;
         let s3_req = super::build_s3_request(input, req);
-        let result = s3.complete_multipart_upload(s3_req).await;
-        let s3_resp = match result {
-            Ok(val) => val,
-            Err(err) => return super::serialize_error(err),
+        let s3 = s3.clone();
+        let fut = async move {
+            let result = s3.complete_multipart_upload(s3_req).await;
+            match result {
+                Ok(s3_resp) => Self::serialize_http(s3_resp.output).unwrap(),
+                Err(err) => super::serialize_error_no_decl(err).unwrap(),
```

Comment:
It's better to propagate the errors instead of aborting current task with a confusing panic. We only use panic when severe errors happen.